internal/api/configapi/repo.go

```diff
@@ -15,41 +15,62 @@ import (
 	"time"
 
 	"github.com/edgelesssys/constellation/v2/internal/kms/storage"
-	"github.com/edgelesssys/constellation/v2/internal/kms/storage/awss3"
-	"github.com/edgelesssys/constellation/v2/internal/kms/uri"
+	"github.com/edgelesssys/constellation/v2/internal/sigstore"
+	"github.com/edgelesssys/constellation/v2/internal/staticupload"
 	"github.com/edgelesssys/constellation/v2/internal/variant"
 )
 
 // AttestationVersionRepo manages (modifies) the version information for the attestation variants.
 type AttestationVersionRepo struct {
-	*awss3.Storage
+	*staticupload.Client
+	cosignPwd []byte // used to decrypt the cosign private key
+	privKey   []byte // used to sign
 }
 
 // NewAttestationVersionRepo returns a new AttestationVersionRepo.
-func NewAttestationVersionRepo(ctx context.Context, cfg uri.AWSS3Config) (*AttestationVersionRepo, error) {
-	s3, err := awss3.New(ctx, cfg)
+func NewAttestationVersionRepo(ctx context.Context, cfg staticupload.Config, cosignPwd, privateKey []byte) (*AttestationVersionRepo, error) {
+	client, err := staticupload.New(ctx, cfg)
 	if err != nil {
 		return nil, fmt.Errorf("failed to create s3 storage: %w", err)
 	}
-	return &AttestationVersionRepo{s3}, nil
+	return &AttestationVersionRepo{client, cosignPwd, privateKey}, nil
 }
 
 // UploadAzureSEVSNP uploads the latest version numbers of the Azure SEVSNP.
 func (a AttestationVersionRepo) UploadAzureSEVSNP(ctx context.Context, versions AzureSEVSNPVersion, date time.Time) error {
-	bt, err := json.Marshal(versions)
+	versionBytes, err := json.Marshal(versions)
 	if err != nil {
 		return err
 	}
 	variant := variant.AzureSEVSNP{}
 	fname := date.Format("2006-01-02-15-04") + ".json"
 
-	err = a.Put(ctx, fmt.Sprintf("%s/%s/%s", attestationURLPath, variant.String(), fname), bt)
+	filePath := fmt.Sprintf("%s/%s/%s", attestationURLPath, variant.String(), fname)
+	err = a.Put(ctx, filePath, versionBytes)
+	if err != nil {
+		return err
+	}
+
+	err = a.signAndUpload(ctx, versionBytes, filePath)
 	if err != nil {
 		return err
 	}
 	return a.addVersionToList(ctx, variant, fname)
 }
 
+// signAndUpload signs the given content and uploads it to the given filePath with the .sig suffix.
+func (a AttestationVersionRepo) signAndUpload(ctx context.Context, content []byte, filePath string) error {
```

Comment:
```suggestion
// uploadSignature signs the given content and uploads the signature to the given filePath with the .sig suffix.
func (a AttestationVersionRepo) uploadSignature(ctx context.Context, content []byte, filePath string) error {
```

Right now the function and it's comment suggest that `content` is also uploaded by the function. However, it is not.

Reply:
uploadSignature also suggests only upload when it really signs and uploads.
how about this:

// signAndUpload signs the given content and uploads the signature to the given filePath with the .sig suffix.

Reply:
maybe use `Sig` instead of `Signature`. your choice.